### PR TITLE
making adjustments for building dash on a windows machine

### PR DESCRIPTION
**Pull request by BSd3v**
This PR is an attempt to ease the build process when using a windows machine:

This build still uses `git bash` to execute the scripts and the build:
* using `rimraf` instead of `rm` for removing files/directories.
* using `mkdirp` instead of `mkdir -p`
* `update_components`:
    * windows kept the `'` which threw off the `all` test.
    * print even with unicode errors
    * dest_dir_map to map components to directories instead of if statement
    * altered the pipe delimited for {} with comma delimited (error with any second+ component)
* added new script `first-build` that will create html components and npm i for the `dash-renderer`

### CODE REVIEW

**Comment by Coding-with-Adam** [comment]
Thank you for this PR, @BSd3v .

File: components/dash-core-components/package.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
     "private::lint.eslint": "eslint src",
     "private::lint.flake8": "flake8 --exclude=dash_core_components,node_modules,venv",
     "private::lint.prettier": "prettier --config .prettierrc src/**/*.js --list-different",
-    "prepublishOnly": "rm -rf lib && babel src --out-dir lib --copy-files --config-file ./.lib.babelrc && rm -rf lib/jl/ lib/*.jl",
+    "prepublishOnly": "rimraf -rf lib && babel src --out-dir lib --copy-files --config-file ./.lib.babelrc && rimraf --glob -rf lib/jl/ lib/*.jl",
```
Comment (alexcjohnson) [inline components/dash-core-components/package.json:22]:
```suggestion
    "prepublishOnly": "rimraf lib && babel src --out-dir lib --copy-files --config-file ./.lib.babelrc && rimraf --glob lib/jl/ lib/*.jl",
```

File: components/dash-html-components/README.md [hunk 2]
```diff
@@ -100,7 +100,7 @@ See the [contributing guide](CONTRIBUTING.md) for guidelines on contributing to
         ```
     2. Cleanup the dist folder (optional)
         ```
-        $ rm -rf dist
+        $ rimraf -rf dist
```
Comment (alexcjohnson) [inline components/dash-html-components/README.md:103]:
```suggestion
        $ rimraf dist
```

File: components/dash-html-components/package.json [hunk 3]
```diff
@@ -13,8 +13,8 @@
   },
   "homepage": "https://github.com/plotly/dash",
   "scripts": {
-    "clean": "rm -rf src/* && mkdir -p src/components",
-    "prebuild": "cd scripts && sh generate-all.sh && cd -",
+    "clean": "rimraf --glob -rf ./src/* && mkdirp ./src/components",
```
Comment (alexcjohnson) [inline components/dash-html-components/package.json:16]:
```suggestion
    "clean": "rimraf --glob ./src/* && mkdirp ./src/components",
```

File: dash/dash-renderer/package.json [hunk 4]
```diff
@@ -4,7 +4,7 @@
   "description": "render dash components in react",
   "main": "build/dash_renderer.min.js",
   "scripts": {
-    "prepublishOnly": "rm -rf lib && babel src --extensions=\".ts,.tsx,.js,.jsx\" --out-dir lib --copy-files",
+    "prepublishOnly": "rimraf -rf lib && babel src --extensions=\".ts,.tsx,.js,.jsx\" --out-dir lib --copy-files",
```
Comment (alexcjohnson) [inline dash/dash-renderer/package.json:7]:
```suggestion
    "prepublishOnly": "rimraf lib && babel src --extensions=\".ts,.tsx,.js,.jsx\" --out-dir lib --copy-files",
```

**Review by alexcjohnson — APPROVED** [review]
💃 That was a slog, thanks for pushing it through @BSd3v!

**Pull request merged**